spec/views/courses/show.html.erb_spec.rb

```diff
@@ -24,4 +24,30 @@
     assert_select "tr>td", text: @lecture.enrollment_key, count: 1
     assert_select "tr>td", text: @lecture.status, count: 1
   end
+
+  context "for students" do
+    before(:each) do
+      login_student
+      @lecture.update(status: "running")
+    end
+
+    it "displays key input form for lectures with a key for not joined students" do
+      render
+      # it's 3 because of the hidden input fields in a form
+      assert_select "form input", count: 3
+      assert_select "form", count: 1
+    end
+
+    it "does not display key input form for lectures without a key for not joined students" do
+      @lecture.update(enrollment_key: nil)
+      render
+      assert_select "form input", count: 1
```

Comment:
schreib hier zur konsistenz halber zum oberen sowas wie "# it's 1 because the button is an input field"